modules/afsocket/afsocket-grammar.ym

```diff
@@ -420,6 +422,25 @@ source_afnetwork_option
 
 source_afsocket_transport
 	: afsocket_transport
+        | KW_TRANSPORT '(' string
+          {
+            Plugin *p;
+            gint context = LL_CONTEXT_SERVER_PROTO;
+
+            p = cfg_find_plugin(configuration, context, $3);
+            if (p)
+              {
+                /* for transports with grammar */
+                if (p->parser)
+                  {
+                    LogProtoServerFactory *sf = cfg_parse_plugin(configuration, p, &@3, last_proto_server_options);
+                    ((AFSocketSourceDriver *) last_driver)->proto_factory = sf;
+                  }
+              }
+            transport_mapper_set_transport(last_transport_mapper, $3);
+            free($3);
```

Comment:
In case of a non-existent transport, a double free occurs here because of our global destructor function (see below).

https://github.com/balabit/syslog-ng/blob/a54587f5ea8959d460ee29064104aae515cb6bed/lib/cfg-grammar.y#L322

Reply:
Fixed.